packages/plugin-ext/src/hosted/browser/hosted-plugin.ts

```diff
@@ -239,50 +239,59 @@ export class HostedPluginSupport {
 
     protected loadQueue: Promise<void> = Promise.resolve(undefined);
     load = debounce(() => this.loadQueue = this.loadQueue.then(async () => {
-        try {
-            await this.progressService.withProgress('', PluginProgressLocation, () => this.doLoad());
-        } catch (e) {
-            console.error('Failed to load plugins:', e);
+        // retry loading if needed due to sporadic "Connection got disposed" error on fast reconnect
```

Comment:
why do we need it, if there is a reconnection, there will be `onDidOpenConnection` event and `load` is called again from above.

Reply:
All this happens and still the first call to `this.pluginPathsService` which is `getHostLogPath` fail with connection got disposed. The second call which comes 0.001 seconds after succeeds (`getHostStoragePath`).

Reply:
@akosyakov what do you think? do you have any alternative approach? or these "connection got disposed" could happen and then retry is right?

Reply:
Do you mean that we call different JSON-RPC services and not all can be reconnected already on `onDidOpenConnection` event? I will need to verify it. I thought that it was not the case.

Reply:
From my experience most of the time when switching WiFis this event is triggered but only after few seconds the service is really working. Before that we get connection got disposed.

Does it make sense?

Reply:
An idea was that the reconnection should be atomic and it is enough to listen to one json-rpc service to be sure that other are reconnected. Maybe there were some breaking change.